nncf/automl/environment/quantization_env.py

```diff
@@ -0,0 +1,574 @@
+                feature['cin'] = m.weight.shape[1]
+                feature['cout'] = m.weight.shape[0]
+                feature['stride'] = m.stride[0]
+                feature['kernel'] = m.kernel_size[0]
+                feature['param'] = np.prod(m.weight.size())
+                feature['ifm_size'] = np.prod(m.input_shape_[-2:]) # H*W
+                feature['prev_action'] = 0.0 # placeholder
+
+            elif isinstance(m, nn.Linear):
+                feature['conv_dw'] = 0.0
+                feature['cin'] = m.in_features
+                feature['cout'] = m.out_features
+                feature['stride'] = 0.0
+                feature['kernel'] = 1.0
+                feature['param'] = np.prod(m.weight.size())
+                feature['ifm_size'] = np.prod(m.input_shape_[-1]) # feature nodes
+                feature['prev_action'] = 0.0 # placeholder
+
+            else:
+                raise NotImplementedError("State embedding extraction of {}".format(m.__class__.__name__))
+
+        elif isinstance(qid, NonWeightQuantizerId):
+            qmod = self.qctrl.all_quantizations[qid]
+            input_shape = qmod.input_shape_
+            output_shape = qmod.output_shape_
+            feature['cin'] = input_shape[1] if len(input_shape) == 4 else input_shape[-1]
+            feature['cout'] = output_shape[1] if len(output_shape) == 4 else output_shape[-1]
+            feature['ifm_size'] = np.prod(input_shape[-2:]) if len(input_shape) == 4 else input_shape[-1]
+            feature['conv_dw'] = 0.0
+            feature['stride'] = 0.0
+            feature['kernel'] = 0.0
+            feature['param'] = 0.0
+            feature['prev_action'] = 0.0
+
+            if len(input_shape) != 4 and len(input_shape) != 2:
+                raise NotImplementedError("A design is required to cater this scenario. Pls. report to maintainer")
+
+        elif isinstance(qid, InputQuantizerId):
+            raise NotImplementedError("InputQuantizerId is supported, quantizer of nncf model input "
+                                      "is of type NonWeightQuantizerId")
+
+        elif isinstance(qid, FunctionQuantizerId):
+            raise NotImplementedError("FunctionQuantizerId is supported, Pls. report to maintainer")
+
+        else:
+            raise ValueError("qid is an instance of unexpected class {}".format(qid.__class__.__name__))
+
+        return pd.Series(feature)
+
+
+    def _evaluate_pretrained_model(self):
+        logger.info("[Q.Env] Evaluating Pretrained Model")
+        self.qctrl.disable_weight_quantization()
+        self.qctrl.disable_activation_quantization()
+
+        with torch.no_grad():
+            self.pretrained_score = self.eval_fn(self.qmodel, self.eval_loader)
+            logger.info("Pretrained Score: {:.2f}".format(self.pretrained_score))
+
+        self.qctrl.enable_weight_quantization()
+        self.qctrl.enable_activation_quantization()
+        self.qmodel.rebuild_graph()
+
+
+    def _adaptbn(self):
```

Comment:
please remove it, as switching to train mode is already inside `self.qctrl.run_batchnorm_adaptation`.
you can directly call `self.qctrl.run_batchnorm_adaptation` only